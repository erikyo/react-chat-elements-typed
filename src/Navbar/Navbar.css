.rce-navbar {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    padding: 5px 10px;
}

.rce-navbar.light {
    background: var(--rce-mbox-light-bg);
}

.rce-navbar.dark {
    background: var(--rce-mbox-dark-bg);
    color: #fff
}

.rce-navbar-item {
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: flex-start;
}

.rce-navbar-item > * {
    display: flex;
    flex-direction: row;
}

.rce-navbar-item > * > * {
    margin-left: 5px;
}
