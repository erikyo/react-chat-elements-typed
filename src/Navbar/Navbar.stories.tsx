import type { Meta, StoryObj } from "@storybook/react";

import Navbar from "./Navbar";

const meta = {
	component: Navbar,
} satisfies Meta<typeof Navbar>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {
	args: {
		left: "left",
		right: "right",
		center: "center",
		style: {},
	},
};

export const Dark: Story = {
	args: {
		left: "left",
		right: "right",
		center: "center",
		type: "dark",
		style: {},
	},
};
