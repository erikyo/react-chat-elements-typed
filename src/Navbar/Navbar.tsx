import type { FC } from "react";
import "./Navbar.css";
import classNames from "classnames";
import type { INavbarProps } from "../types";

const Navbar: FC<INavbarProps> = ({ type = "light", ...props }) => {
	return (
		<div className={classNames("rce-navbar", type, props.className)}>
			<div className="rce-navbar-item rce-navbar-item__left">{props.left}</div>
			<div className="rce-navbar-item rce-navbar-item__center">
				{props.center}
			</div>
			<div className="rce-navbar-item rce-navbar-item__right">
				{props.right}
			</div>
		</div>
	);
};

export default Navbar;
