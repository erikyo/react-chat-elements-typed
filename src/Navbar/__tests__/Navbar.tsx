import { describe, expect, it } from "vitest";
import React from "react";
import Navbar from "../Navbar";
import { render } from "@testing-library/react";

describe("Avatar component", () => {
	it("should render without issues", () => {
		const component = render(<Navbar />);
		expect(component.container).toBeTruthy();
		expect(component.container).toMatchSnapshot();
	});
});
