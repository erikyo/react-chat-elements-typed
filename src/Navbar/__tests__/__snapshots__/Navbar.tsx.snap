// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Avatar component > should render without issues 1`] = `
<div>
  <div
    class="rce-navbar default"
    style="width: 100%;"
  >
    <div
      class="rce-navbar-item rce-navbar-item__left"
    />
    <div
      class="rce-navbar-item rce-navbar-item__center"
    />
    <div
      class="rce-navbar-item rce-navbar-item__right"
    />
  </div>
</div>
`;
